photoprism/docker-compose.yaml

```diff
@@ -0,0 +1,132 @@
+version: '3.5'
+
+# Note:
+# - Running PhotoPrism on a server with less than 4 GB of swap space or setting a memory/swap limit can cause unexpected
+#   restarts ("crashes"), for example, when the indexer temporarily needs more memory to process large files.
+#
+# Setup Guides:
+# - https://docs.photoprism.app/getting-started/docker-compose/
+# - https://docs.photoprism.app/getting-started/raspberry-pi/
+# - https://www.photoprism.app/kb/activation
+#
+# Troubleshooting Checklists:
+# - https://docs.photoprism.app/getting-started/troubleshooting/
+# - https://docs.photoprism.app/getting-started/troubleshooting/docker/
+# - https://docs.photoprism.app/getting-started/troubleshooting/mariadb/
+#
+# CLI Commands:
+# - https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface
+#
+# All commands may have to be prefixed with "sudo" when not running as root.
+# This will point the home directory shortcut ~ to /root in volume mounts.
+
+volumes:
+  photoprism-infra:
+  photoprism-mariadb:
+
+services:
+  photoprism:
+    image: photoprism/photoprism:${PHOTOPRISM_VERSION}
+    ## If the service gets stuck in a restart loop, this points to a memory, filesystem, network, or database issue:
+    ## https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
+    restart: unless-stopped
+    stop_grace_period: 10s
+    depends_on:
+      - mariadb
+# are these security options safe? or do i want `no-new-privileges:true` and `cap_drop:      - ALL`?
```

Comment:
no-new-privileges:true is always a good default, as long as you test that it is working, it will mean it cannot get more privilges than you've defined here. cap_drop: all is a more drastic approach that drops all privileges and you need to add every single one it needs; this usually breaks the app, but you just test it adding more capapilities until it works. 

I'm guessing you copied these from the photoprism config, which is fine, you might try using the no-new-privilges at least though.

Reply:
I changed to `no-new-priveleges:true` and it seems to work fine.